### PR TITLE
HPCC-19145 Fix some JSON output issues.

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
NB: Not ready for merging yet.

**Comment by jakesmith** [comment]
@afishbeck - please review what's here + look at the JSON IXmlWriterExt issue:
```
// JCSMORE - json string coming back from IXmlWriterExt is always rootless the moment, so workaround it by supplying ptr_noRoot to reader
// writer should be fixed.
```
which is causing the "Response" to not be in a object { }

**Comment by jakesmith** [comment]
@afishbeck - https://track.hpccsystems.com/browse/HPCC-18831 will address the rootless issue.
I've linked this issue and made a comment there about a dependent change that will need to be made when that change is made.

The above smoketest failure is unrelated - looks like it's related to an upmerged change from this issue: https://track.hpccsystems.com/browse/HPCC-19874

**Comment by jakesmith** [comment]
Removed namescope from "Row".

@afishbeck - please review

**Comment by AttilaVamos** [comment]
Master is fixed (related to xmlread error) and this PR is rescheduled.

**Review by afishbeck — APPROVED** [review]
@jakesmith looks good.

**Comment by jakesmith** [comment]
@afishbeck - there was a merge conflict that needed solving, now resolved. Please double check.

**Comment by AttilaVamos** [comment]
I don't think that faulted  DigiSignUnitTest is related to this PR. Re-schedule and let us see.

**Comment by jakesmith** [comment]
@AttilaVamos - the 'DigiSignUnitTest' is unrelated to this PR - but should be looked at.
Can you get some details of where it crashed ( a stack ) , so that @RussWhitehead can look at?

**Comment by jakesmith** [comment]
@AttilaVamos - well re-scheduled run might not crash - but that doesn't mean there's not a temperamental issue that should be investigated. Is there a core/logs with stack for the segfauilt ?

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -6376,9 +6378,11 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
             else
             {
+                responseWriter->outputBeginArray("dfs:Row");
```
Comment (afishbeck) [inline common/remote/sockfile.cpp]: @jakesmith technically "dfs:Row" is not correct.  The contents of Row are defined by the ECL Record specific schema, not by the schema identified by the unique id "urn:hpcc:dfs".  I would only add the dfs namepace to the EOG boolean.

**Review by afishbeck — CHANGES_REQUESTED** [review]

**Comment by AttilaVamos** [comment]
@jakesmith no core file and no more logs, yet.

**Comment by jakesmith** [comment]
@afishbeck - it looks like my merge conflict was to an old version of my branch and lost the "dfs:Row" changes in the process.

Have now pushed a commit to fix and have re-tested.
Please check and I'll squash the new commit.

**Review by afishbeck — APPROVED** [review]
@jakesmith looks good

**Pull request merged**